### PR TITLE
fix(TowerWorkflow): correct task duration

### PR DESCRIPTION
**Pull request by fevac**
### Fixed

- Duration is retrieved from Tower API in milliseconds instead of seconds. Converting to seconds to report duration correctly.

### Changed

- Updated code owners to infrastructure

### How to prepare for test
- [x] ssh to hasta (depending on type of change)
- [x] activate stage: `us`
- [x] request trailblazer-stage on hasta: `paxa`
- [x] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b fix_time_tower -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [x] Run `trailblazer scan` with at least one pending rnafusion case. Check that duration is correctly reported in minutes. 
image

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: trailblazer/apps/tower/models.py [hunk 1]
```diff
@@ -64,8 +64,9 @@ class Config:
         validate_all = True
 
     @validator("duration")
-    def set_duration(cls, duration) -> str:
-        return duration or 0
+    def set_duration(cls, duration) -> int:
```
Comment (ivadym) [inline trailblazer/apps/tower/models.py:67]:
```suggestion
    def set_duration(cls, duration: Optional[int]) -> int:
```

**Review by ivadym — APPROVED** [review]
Nice! ⌚

**Review by karlnyr — APPROVED** [review]

**Comment by fevac** [comment]
thanks @ivadym and @karlnyr for the review!

File: trailblazer/apps/tower/models.py [hunk 2]
```diff
@@ -64,8 +64,9 @@ class Config:
         validate_all = True
 
     @validator("duration")
-    def set_duration(cls, duration) -> str:
-        return duration or 0
+    def set_duration(cls, duration: Optional[int]) -> int:
+        """Convert milliseconds to seconds or return 0 if empty."""
+        return round(duration / 1000) if duration else 0
```
Comment (henrikstranneheim) [inline trailblazer/apps/tower/models.py:69]:
```suggestion
        return round(duration / SCALE_TO_MILLISEC) if duration else 0
```
And define the constant perhaps in this module as it is quite specific.
`SCALE_TO_MILLISEC: int = 1_000`

**Review by henrikstranneheim — APPROVED** [review]
👍

**Comment by fevac** [comment]
- Deployed in stage

image



- Deployed in prod

image

**Pull request merged**